From b3f6951df462181daac258df152b58858fb172f8 Mon Sep 17 00:00:00 2001
From: Suman Tatiraju <sumant@codeaurora.org>
Date: Tue, 15 Dec 2015 07:49:41 -0700
Subject: [PATCH 803/959] msm: kgsl: Fix direct references to HZ

Make the various timeout values HZ agnostic by using the proper
macros and values instead.

Mot-CRs-fixed: IKKRNBSP-3775

Change-Id: I708cd491f593782f0172cd7d2cca058cd41044a5
Signed-off-by: Suman Tatiraju <sumant@codeaurora.org>
(cherry picked from commit 1b669de73dbb4d770c9fabb75ff0b8dd0c9e3895)
Reviewed-on: http://gerrit.mot.com/833999
SME-Granted: SME Approvals Granted
SLTApproved: Slta Waiver <sltawvr@motorola.com>
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Fred Fettinger <fettinge@motorola.com>
Reviewed-by: Joseph Swantek <jswantek@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 Documentation/devicetree/bindings/gpu/adreno.txt | 2 +-
 arch/arm/boot/dts/qcom/apq8084-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8226-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8610-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8909-gpu.dtsi          | 3 +--
 arch/arm/boot/dts/qcom/msm8916-gpu.dtsi          | 3 +--
 arch/arm/boot/dts/qcom/msm8929-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8939-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8939-v3.0-gpu.dtsi     | 2 +-
 arch/arm/boot/dts/qcom/msm8952-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8974-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8976-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8992-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8994-gpu.dtsi          | 2 +-
 arch/arm/boot/dts/qcom/msm8x26w-gpu.dtsi         | 2 +-
 drivers/gpu/msm/adreno.c                         | 2 +-
 drivers/gpu/msm/adreno_dispatch.c                | 3 ++-
 drivers/gpu/msm/adreno_profile.c                 | 2 +-
 drivers/gpu/msm/kgsl_device.h                    | 2 --
 drivers/gpu/msm/kgsl_pwrctrl.c                   | 4 ++--
 20 files changed, 21 insertions(+), 24 deletions(-)

diff --git a/Documentation/devicetree/bindings/gpu/adreno.txt b/Documentation/devicetree/bindings/gpu/adreno.txt
index 1db29b8070d..fedf1ec9f9b 100644
--- a/Documentation/devicetree/bindings/gpu/adreno.txt
+++ b/Documentation/devicetree/bindings/gpu/adreno.txt
@@ -94,7 +94,7 @@ Optional Properties:
 			   bus width and actual bus transactions.
 - qcom,gpubw-dev:	   a phandle to a device representing bus bandwidth requirements
 			   (see devdw.txt)
-- qcom,idle-timeout:	   This property represents the time in microseconds for idle timeout.
+- qcom,idle-timeout:	   This property represents the time in milliseconds for idle timeout.
 - qcom,chipid:		   If it exists this property is used to replace
 			   the chip identification read from the GPU hardware.
 			   This is used to override faulty hardware readings.
diff --git a/arch/arm/boot/dts/qcom/apq8084-gpu.dtsi b/arch/arm/boot/dts/qcom/apq8084-gpu.dtsi
index a3d8090e804..5c8e896e051 100644
--- a/arch/arm/boot/dts/qcom/apq8084-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/apq8084-gpu.dtsi
@@ -30,7 +30,7 @@
 
 		qcom,initial-pwrlevel = <3>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		qcom,pm-qos-active-latency = <501>;
diff --git a/arch/arm/boot/dts/qcom/msm8226-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8226-gpu.dtsi
index 348a7582dd9..220eff3f4a3 100644
--- a/arch/arm/boot/dts/qcom/msm8226-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8226-gpu.dtsi
@@ -24,7 +24,7 @@
 
 		qcom,initial-pwrlevel = <1>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 		qcom,clk-map = <0x00000016>; /* KGSL_CLK_CORE | KGSL_CLK_IFACE | KGSL_CLK_MEM_IFACE */
 
diff --git a/arch/arm/boot/dts/qcom/msm8610-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8610-gpu.dtsi
index 9e5a096db39..2ac5e5959e1 100644
--- a/arch/arm/boot/dts/qcom/msm8610-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8610-gpu.dtsi
@@ -24,7 +24,7 @@
 
 		qcom,initial-pwrlevel = <1>;
 
-		qcom,idle-timeout = <8>; /* <HZ/12> */
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 		qcom,clk-map = <0x000005E>; /* KGSL_CLK_CORE |
 			KGSL_CLK_IFACE | KGSL_CLK_MEM | KGSL_CLK_MEM_IFACE |
diff --git a/arch/arm/boot/dts/qcom/msm8909-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8909-gpu.dtsi
index ba881d07615..131b27fd604 100644
--- a/arch/arm/boot/dts/qcom/msm8909-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8909-gpu.dtsi
@@ -52,8 +52,7 @@
 
 		qcom,initial-pwrlevel = <1>;
 
-		/* Idle Timeout = HZ/12 */
-		qcom,idle-timeout = <8>;
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/*
diff --git a/arch/arm/boot/dts/qcom/msm8916-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8916-gpu.dtsi
index 9a5dd19259e..a0e8e94a50e 100644
--- a/arch/arm/boot/dts/qcom/msm8916-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8916-gpu.dtsi
@@ -29,8 +29,7 @@
 
 		qcom,initial-pwrlevel = <1>;
 
-		/* Idle Timeout = HZ/12 */
-		qcom,idle-timeout = <8>;
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/*
diff --git a/arch/arm/boot/dts/qcom/msm8929-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8929-gpu.dtsi
index 2b295e3ceec..09dfcefe51d 100644
--- a/arch/arm/boot/dts/qcom/msm8929-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8929-gpu.dtsi
@@ -49,7 +49,7 @@
 
 		qcom,initial-pwrlevel = <1>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/* Clocks = KGSL_CLK_CORE | KGSL_CLK_IFACE | KGSL_CLK_MEM |
diff --git a/arch/arm/boot/dts/qcom/msm8939-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8939-gpu.dtsi
index fa4463afe2f..d87b28b8965 100644
--- a/arch/arm/boot/dts/qcom/msm8939-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8939-gpu.dtsi
@@ -51,7 +51,7 @@
 
 		qcom,initial-pwrlevel = <2>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/* Clocks = KGSL_CLK_CORE | KGSL_CLK_IFACE | KGSL_CLK_MEM |
diff --git a/arch/arm/boot/dts/qcom/msm8939-v3.0-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8939-v3.0-gpu.dtsi
index 86288150baa..3e645de9624 100644
--- a/arch/arm/boot/dts/qcom/msm8939-v3.0-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8939-v3.0-gpu.dtsi
@@ -50,7 +50,7 @@
 
 		qcom,initial-pwrlevel = <2>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/* Clocks = KGSL_CLK_CORE | KGSL_CLK_IFACE | KGSL_CLK_MEM |
diff --git a/arch/arm/boot/dts/qcom/msm8952-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8952-gpu.dtsi
index ff1ac852f63..b7a0995769c 100644
--- a/arch/arm/boot/dts/qcom/msm8952-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8952-gpu.dtsi
@@ -58,7 +58,7 @@
 
 		qcom,initial-pwrlevel = <3>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/* Clocks = KGSL_CLK_CORE | KGSL_CLK_IFACE | KGSL_CLK_MEM |
diff --git a/arch/arm/boot/dts/qcom/msm8974-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8974-gpu.dtsi
index 34774b597e6..d777f1a2597 100644
--- a/arch/arm/boot/dts/qcom/msm8974-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8974-gpu.dtsi
@@ -24,7 +24,7 @@
 
 		qcom,initial-pwrlevel = <1>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 		qcom,clk-map = <0x0000006>; //KGSL_CLK_CORE | KGSL_CLK_IFACE
 
diff --git a/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
index e29b75c19a4..423ffd44155 100644
--- a/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
@@ -51,7 +51,7 @@
 
 		qcom,initial-pwrlevel = <5>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		qcom,pm-qos-active-latency = <401>;
diff --git a/arch/arm/boot/dts/qcom/msm8992-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8992-gpu.dtsi
index 6534355281f..2b36fe4da85 100644
--- a/arch/arm/boot/dts/qcom/msm8992-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8992-gpu.dtsi
@@ -46,7 +46,7 @@
 
 		qcom,initial-pwrlevel = <4>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		/*
diff --git a/arch/arm/boot/dts/qcom/msm8994-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8994-gpu.dtsi
index 3d63cf8d207..fc08d975b3f 100644
--- a/arch/arm/boot/dts/qcom/msm8994-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8994-gpu.dtsi
@@ -57,7 +57,7 @@
 
 		qcom,initial-pwrlevel = <2>;
 
-		qcom,idle-timeout = <8>; //<HZ/12>
+		qcom,idle-timeout = <80>; //msecs
 		qcom,strtstp-sleepwake;
 
 		qcom,pm-qos-active-latency = <501>;
diff --git a/arch/arm/boot/dts/qcom/msm8x26w-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8x26w-gpu.dtsi
index 06e06c128c9..9dea811a741 100644
--- a/arch/arm/boot/dts/qcom/msm8x26w-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8x26w-gpu.dtsi
@@ -12,5 +12,5 @@
 
 &msm_gpu {
 	qcom,initial-pwrlevel = <2>;
-	qcom,idle-timeout = <5>;
+	qcom,idle-timeout = <50>; //msecs
 };
diff --git a/drivers/gpu/msm/adreno.c b/drivers/gpu/msm/adreno.c
index c5eff53e1fb..2a2ce98104c 100644
--- a/drivers/gpu/msm/adreno.c
+++ b/drivers/gpu/msm/adreno.c
@@ -1182,7 +1182,7 @@ static int adreno_of_get_pdata(struct platform_device *pdev)
 
 	if (of_property_read_u32(pdev->dev.of_node, "qcom,idle-timeout",
 		&pdata->idle_timeout))
-		pdata->idle_timeout = HZ/12;
+		pdata->idle_timeout = 80;
 
 	pdata->strtstp_sleepwake = of_property_read_bool(pdev->dev.of_node,
 						"qcom,strtstp-sleepwake");
diff --git a/drivers/gpu/msm/adreno_dispatch.c b/drivers/gpu/msm/adreno_dispatch.c
index 91150db3150..6b6d4ce2d15 100644
--- a/drivers/gpu/msm/adreno_dispatch.c
+++ b/drivers/gpu/msm/adreno_dispatch.c
@@ -437,7 +437,8 @@ static struct kgsl_cmdbatch *_get_cmdbatch(struct adreno_context *drawctxt)
 		 * it hasn't already been started
 		 */
 		if (!cmdbatch->timeout_jiffies) {
-			cmdbatch->timeout_jiffies = jiffies + 5 * HZ;
+			cmdbatch->timeout_jiffies =
+				jiffies + msecs_to_jiffies(5000);
 			mod_timer(&cmdbatch->timer, cmdbatch->timeout_jiffies);
 		}
 
diff --git a/drivers/gpu/msm/adreno_profile.c b/drivers/gpu/msm/adreno_profile.c
index 5476f9892f8..5d5eedcaa79 100644
--- a/drivers/gpu/msm/adreno_profile.c
+++ b/drivers/gpu/msm/adreno_profile.c
@@ -984,7 +984,7 @@ static ssize_t profile_pipe_print(struct file *filep, char __user *ubuf,
 
 		mutex_unlock(&device->mutex);
 		set_current_state(TASK_INTERRUPTIBLE);
-		schedule_timeout(HZ / 10);
+		schedule_timeout(msecs_to_jiffies(100));
 		mutex_lock(&device->mutex);
 
 		if (signal_pending(current)) {
diff --git a/drivers/gpu/msm/kgsl_device.h b/drivers/gpu/msm/kgsl_device.h
index 1b677c0c759..368e422d59b 100644
--- a/drivers/gpu/msm/kgsl_device.h
+++ b/drivers/gpu/msm/kgsl_device.h
@@ -33,8 +33,6 @@
 #define KGSL_TIMEOUT_DEFAULT        0xFFFFFFFF
 #define KGSL_TIMEOUT_PART           50 /* 50 msec */
 
-#define FIRST_TIMEOUT (HZ / 2)
-
 #define KGSL_IOCTL_FUNC(_cmd, _func) \
 	[_IOC_NR((_cmd))] = \
 		{ .cmd = (_cmd), .func = (_func) }
diff --git a/drivers/gpu/msm/kgsl_pwrctrl.c b/drivers/gpu/msm/kgsl_pwrctrl.c
index f50e6d788a3..7bf7469e2b7 100644
--- a/drivers/gpu/msm/kgsl_pwrctrl.c
+++ b/drivers/gpu/msm/kgsl_pwrctrl.c
@@ -47,7 +47,7 @@
 #define MAX_UDELAY		2000
 
 /* Number of jiffies for a full thermal cycle */
-#define TH_HZ			20
+#define TH_HZ			(HZ/5)
 
 #define KGSL_MAX_BUSLEVELS	20
 
@@ -1564,7 +1564,7 @@ int kgsl_pwrctrl_init(struct kgsl_device *device)
 
 	pwr->power_flags = 0;
 
-	pwr->interval_timeout = pdata->idle_timeout;
+	pwr->interval_timeout = msecs_to_jiffies(pdata->idle_timeout);
 	pwr->strtstp_sleepwake = pdata->strtstp_sleepwake;
 
 	if (kgsl_property_read_u32(device, "qcom,pm-qos-active-latency",
-- 
2.11.0

